src/dotenv/main.py

```diff
@@ -135,11 +171,13 @@ def set_as_environment_variables(self, override=False):
                 if isinstance(k, text_type) or isinstance(v, text_type):
                     k = k.encode('ascii')
                     v = v.encode('ascii')
-            os.environ[k] = v
+            if not TYPE_CHECKING or sys.version_info >= (3, 0):
```

Comment:
I think this obscures the code too much. I would probably just ignore the type warning for the line below.